### ISSUE TITLE
Convergence testing best practices

### ISSUE THREAD

**Issue by gnadathur** [pytorch/torchtitan#648]
Config for convergence testing is unclear. Create a best practice config in torchtitan for convergence testing.

**Comment by lessw2020** [comment]
Two core aspects here:
1 - ensure determinism (assuming these runs are to compare various dtypes, etc).  utils.set_determinism should handle this but need to confirm cublas setting has landed from Wei.
2 - LR warmup is not being properly set.  Basic schedule optimization is to pre-run while tracking L1 gradient norms, and feed that to Aaron's refined scheduler based on:
https://arxiv.org/abs/2310.07831

**Comment by awgu** [comment]
@lessw2020 I am not sure if convergence testing should have determinism enabled because ultimately that will not be the setting you train with (and care to converge).

I think determinism is super useful for debugging and understand where numeric differences come from, but I think that for convergence testing, running with "prod" settings makes more sense to me.

**Comment by fduwjj** [comment]
I guess for torchTitan, we really cannot call it "convergence" right? since we never wait the training to finish and a full convergence is achieved and we tested on test benchmark to validate the accuracy of the model. To me that is what "convergence" means. Currently we only show "Hey the loss curve is going down".

**Comment by tianyu-l** [comment]
> we really cannot call it "convergence" right?

Technically yes. Maybe we can call them "loss converging" tests. I don't think it would cause confusion, so it should be fine as long as we don't say wrong things (convergence is wrong).

**Comment by wconstab** [comment]
Action:
Document what are the guidelines for doing runs that are used to show a 'config' is working correctly. 

 - what settings to use?  (seed? non-determinism)
 - how many steps?  ...
